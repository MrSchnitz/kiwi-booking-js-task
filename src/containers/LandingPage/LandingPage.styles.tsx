import styled from "styled-components";

export const LandingPageWrapper = styled.div`
  display: flex;

  justify-content: center;
  align-items: center;

  padding-top: 52px;
  width: 100%;
  min-height: 100vh;

  background-image: url("https://images.skypicker.com/?image=https%3A%2F%2Fwww.kiwi.com%2Fimages%2Fhero%2Fhero-xxLargeDesktop-ua2%402x.webp&width=2560&height=400&quality=75&dpr=2");
  background-size: cover;
  background-position: top;
`;

export const LandingPageSearchForm = styled.div`
  border-radius: 8px;
  padding: 24px;

  background: rgb(255, 255, 255);

  box-shadow: rgb(37 42 49 / 30%) 0 4px 12px 0;
`;
