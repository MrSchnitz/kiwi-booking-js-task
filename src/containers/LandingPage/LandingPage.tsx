import React from "react";
import {
  LandingPageSearchForm,
  LandingPageWrapper,
} from "./LandingPage.styles";
import { Layout, LayoutColumn } from "@kiwicom/orbit-components";
import SearchForm from "../../components/SearchForm/SearchForm";

export default React.memo(function HomePage() {
  return (
    <LandingPageWrapper>
      <Layout type="MMB">
        <LayoutColumn>
          <LandingPageSearchForm>
            <SearchForm />
          </LandingPageSearchForm>
        </LayoutColumn>
      </Layout>
    </LandingPageWrapper>
  );
});
