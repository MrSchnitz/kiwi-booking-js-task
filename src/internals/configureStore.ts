import { combineReducers, configureStore } from "@reduxjs/toolkit";
import rootSaga, { mainReducers } from "./rootState";
import createSagaMiddleware from "redux-saga";

export function configureAppStore() {
  const reduxSagaMonitorOptions = {};
  const sagaMiddleware = createSagaMiddleware(reduxSagaMonitorOptions);

  // Create the store with saga middleware and router middleware
  const middlewares = [sagaMiddleware];

  const store = configureStore({
    reducer: combineReducers({ ...mainReducers }),
    middleware: (getDefaultMiddleware) => [
      ...getDefaultMiddleware(),
      ...middlewares,
    ],
    devTools: process.env.NODE_ENV !== "production",
  });

  // Add all combined Sagas to middleware
  sagaMiddleware.run(rootSaga);

  return store;
}
