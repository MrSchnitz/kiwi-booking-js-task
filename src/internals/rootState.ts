import { all } from "redux-saga/effects";

export interface RootState {

}

export const mainReducers = {

};


export default function* rootSaga() {
    yield all([]);
}
