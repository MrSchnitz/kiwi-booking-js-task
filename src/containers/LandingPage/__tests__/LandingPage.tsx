import { render } from "@testing-library/react";
import LandingPage from "../LandingPage";
import { Provider } from "react-redux";
import { configureAppStore } from "../../../internals/configureStore";
import { MemoryRouter } from "react-router-dom";

describe("LandingPage", () => {
  const store = configureAppStore();

  it("should render basic component", () => {
    const { container } = render(
      <Provider store={store}>
        <MemoryRouter>
          <LandingPage />
        </MemoryRouter>
      </Provider>
    );

    expect(container).toMatchSnapshot();
  });
});
