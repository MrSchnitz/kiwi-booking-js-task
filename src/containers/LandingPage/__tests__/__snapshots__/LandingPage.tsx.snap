// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingPage should render basic component 1`] = `
<div>
  <div
    class="sc-bczRLJ cDvcuM"
  >
    <div
      class="Grid__StyledGrid-sc-9eqnnl-0 fgKasZ Layout__StyledLayout-sc-2e7ifp-0 jMiYaA"
    >
      <main
        class="LayoutColumn__StyledColumn-sc-7eesuv-0 imJxEk"
      >
        <div
          class="sc-gsnTZi bkEWPv"
        >
          <div
            class="sc-dkzDqf hAPQBN"
          >
            <div
              class="Heading__StyledHeading-sc-rrd5d-0 gurnTJ"
            >
              Search for flights
            </div>
            <div
              class="Stack__StyledStack-sc-oaff2v-0 hFnGib"
            >
              <div
                class="Stack__StyledStack-sc-oaff2v-0 jrgtzq"
              >
                <div
                  class="sc-eCYdqJ bElTwV"
                >
                  <div
                    class="InputField__Field-sc-jjv066-0 koAYkz"
                  >
                    <div
                      class="InputField__InputContainer-sc-jjv066-2 bFmIrl"
                    >
                      <div
                        class="InputField__Prefix-sc-jjv066-4 kUJKxj"
                      >
                        From
                      </div>
                      <input
                        aria-labelledby="1"
                        class="InputField__Input-sc-jjv066-6 kPMZTv"
                        data-state="ok"
                        id="1"
                        type="text"
                        value=""
                      />
                      <div
                        class="InputField__FakeInput-sc-jjv066-1 dmsAbX"
                      />
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="Stack__StyledStack-sc-oaff2v-0 jrgtzq"
              >
                <div
                  class="sc-eCYdqJ bElTwV"
                >
                  <div
                    class="InputField__Field-sc-jjv066-0 koAYkz"
                  >
                    <div
                      class="InputField__InputContainer-sc-jjv066-2 bFmIrl"
                    >
                      <div
                        class="InputField__Prefix-sc-jjv066-4 kUJKxj"
                      >
                        To
                      </div>
                      <input
                        aria-labelledby="2"
                        class="InputField__Input-sc-jjv066-6 kPMZTv"
                        data-state="ok"
                        id="2"
                        type="text"
                        value=""
                      />
                      <div
                        class="InputField__FakeInput-sc-jjv066-1 dmsAbX"
                      />
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="react-datepicker-wrapper"
              >
                <div
                  class="react-datepicker__input-container"
                >
                  <div
                    class="InputField__Field-sc-jjv066-0 koAYkz"
                  >
                    <div
                      class="InputField__InputContainer-sc-jjv066-2 bFmIrl"
                    >
                      <div
                        class="InputField__Prefix-sc-jjv066-4 kUJKxj"
                      >
                        Departure
                      </div>
                      <input
                        aria-labelledby="3"
                        class="InputField__Input-sc-jjv066-6 kPMZTv"
                        data-state="ok"
                        id="3"
                        placeholder="Set date..."
                        type="text"
                        value=""
                      />
                      <div
                        class="InputField__FakeInput-sc-jjv066-1 dmsAbX"
                      />
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="react-datepicker-wrapper"
              >
                <div
                  class="react-datepicker__input-container"
                >
                  <div
                    class="InputField__Field-sc-jjv066-0 koAYkz"
                  >
                    <div
                      class="InputField__InputContainer-sc-jjv066-2 bFmIrl"
                    >
                      <div
                        class="InputField__Prefix-sc-jjv066-4 kUJKxj"
                      >
                        Return
                      </div>
                      <input
                        aria-labelledby="4"
                        class="InputField__Input-sc-jjv066-6 kPMZTv"
                        data-state="ok"
                        id="4"
                        placeholder="Set date..."
                        type="text"
                        value=""
                      />
                      <div
                        class="InputField__FakeInput-sc-jjv066-1 dmsAbX"
                      />
                    </div>
                  </div>
                </div>
              </div>
              <button
                class="ButtonPrimitive__StyledButtonPrimitive-sc-1lbd19y-0 jhJQUb"
                disabled=""
                type="button"
              >
                <div
                  class="ButtonPrimitiveContent__StyledButtonPrimitiveContent-sc-1r81o9a-0 brimGn"
                >
                  <div
                    class="ButtonPrimitiveContentChildren__StyledButtonPrimitiveContentChildren-sc-1m4y8u8-0 jaHPWv"
                  >
                    Search
                  </div>
                </div>
              </button>
            </div>
          </div>
        </div>
      </main>
    </div>
  </div>
</div>
`;
